src/Biogeochemistry.jl

```diff
@@ -0,0 +1,138 @@
+module Biogeochemistry
+
+using Oceananigans.Grids: Center, xnode, ynode, znode
+using Oceananigans.Advection: div_Uc, CenteredSecondOrder
+using Oceananigans.Utils: tupleit
+using Oceananigans.Fields: Field
+using Oceananigans.Architectures: device, architecture
+using Oceananigans.Utils: launch!
+using KernelAbstractions: @kernel
+
+import Oceananigans.Fields: location, CenterField
+import Oceananigans.Forcings: regularize_forcing, maybe_constant_field
+
+#####
+##### Generic fallbacks for biogeochemistry
+#####
+
+"""
+Update tendencies.
+
+Called at the end of calculate_tendencies!
+"""
+update_tendencies!(bgc, model) = nothing
+
+"""
+Update tracer tendencies.
+
+Called at the end of calculate_tendencies!
+"""
+update_biogeochemical_state!(bgc, model) = nothing
+
+@inline biogeochemical_drift_velocity(bgc, val_tracer_name) = nothing
+@inline biogeochemical_advection_scheme(bgc, val_tracer_name) = nothing
+@inline biogeochemical_auxiliary_fieilds(bgc) = NamedTuple()
+
+#####
+##### Default (discrete form) biogeochemical source
+#####
+
+abstract type AbstractBiogeochemistry end
+
+@inline function biogeochemistry_rhs(i, j, k, grid, bgc, val_tracer_name::Val{tracer_name}, clock, fields) where tracer_name
+    U_drift = biogeochemical_drift_velocity(bgc, val_tracer_name)
+    scheme = biogeochemical_advection_scheme(bgc, val_tracer_name)
+    src = biogeochemical_transition(i, j, k, grid, bgc, val_tracer_name, clock, fields)
```

Comment:
what is a `biogeochemical_transition`? Shall we put a comment?

Reply:
`biogeochemical_transition` is the user defined biogeochemical forcing, I can add a docstring or we could just change to line to:
```julia
src = bgc(i, j, k, grid, val_tracer_name, clock, fields)
```
Since that is what it does anyway?

Reply:
We did need this, for continuous form it does something else. I'll add back and add comment.

Reply:
As below, it turns out we do need this for the continous form:

https://github.com/CliMA/Oceananigans.jl/blob/85920168d1e0214ba9586f9bfa492a8b8aaefd32/src/Biogeochemistry.jl#L113-L126

So I've added a commend explaining what it does